addons/udes_transport_management/models/transport.py

```diff
@@ -59,29 +59,38 @@ def get_info(self, **kwargs):
         return res
 
     def prepare_transport_info(self, **kwargs):
-        """Prepare transport info method that can be override in order
+        """Prepare transport info method that can be overridden in order
         to be able to extend it in other modules"""
-        self.ensure_one()
-        transport_info = [
-            {
-                "label": "Trailer Number",
-                "value": self.vehicle_sequence or None,
-                "lookup_key": "trailer_number",
-            },
-            {
-                "label": "Trailer Unit ID",
-                "value": self.vehicle_description or None,
-                "lookup_key": "trailer_unit_id",
-            },
-            {
-                "label": "Vehicle Registration",
-                "value": self.license_plate or None,
-                "lookup_key": "vehicle_registration",
-            },
-            {
-                "label": "Driver Name",
-                "value": self.driver_name or None,
-                "lookup_key": "driver_name",
-            }
-        ]
-        return transport_info
+        blank_transport = (
+                {
+                    "label": _("Trailer Number"),
+                    "value": None,
+                    "lookup_key": "trailer_number",
+                },
+                {
+                    "label": _("Trailer Unit ID"),
+                    "value": None,
+                    "lookup_key": "trailer_unit_id",
+                },
+                {
+                    "label": _("Vehicle Registration"),
+                    "value": None,
+                    "lookup_key": "vehicle_registration",
+                },
+                {
+                    "label": _("Driver Name"),
+                    "value": None,
+                    "lookup_key": "driver_name",
+                },
+            )
+        result = []
+        for transport in self:
+            transport_info = blank_transport.copy()
```

Comment:
Ignore this if i have missed something,

copying a tuple is this a method in odoo, or some other module?
I don't understand how this gets updated?
We have a tuple of dicts, then we pass the key to the tuple and it chooses the relevant dict and sets the value. 

We could do 
lambda transport: (
            {
                "label": _("Trailer Number"),
                "value": transport.vehicle_sequence or None,
                "lookup_key": "trailer_number",
            },
            {
                "label": _("Trailer Unit ID"),
                "value": transport.vehicle_description or None,
                "lookup_key": "trailer_unit_id",
            },
            {
                "label": _("Vehicle Registration"),
                "value": transport.license_plate or None,
                "lookup_key": "vehicle_registration",
            },
            {
                "label": _("Driver Name"),
                "value": transport.driver_name or None,
                "lookup_key": "driver_name",
            },
        )

Then do something like result =[t(transport) for transport in self]?

Their maybe something nicer. 

Could also consider updating the unit tests as they don't currently run

Reply:
About the tuple, I got confused and thought I could access it like in workflows. I am converting it into a list of dictionaries now. About the approach you suggested, it seems nice but I don't see how having a lambda can be used with no records in self if we need to pass a record to lambda in order to create a transport tuple or list with None values. Basically, I don't see a better way of doing this because lambda cannot be used for the empty case.